spec/controllers/api/v0x0/application_controller_spec.rb

```diff
@@ -0,0 +1,67 @@
+RSpec.describe ApplicationController, :type => :request do
+  include SpecHelper
+
+  let(:tenant)          { Tenant.create!(:external_tenant => external_tenant) }
+  let(:source_type)     { SourceType.create!(:name => "openshift", :product_name => "OpenShift", :vendor => "Red Hat") }
+  let(:source)          { Source.create!(:source_type_id => source_type.id.to_s, :tenant_id => tenant.id , :name => "abc", :uid => "123") }
+  let(:external_tenant) { rand(1000).to_s }
+  let(:identity)        { Base64.encode64({'identity' => { 'account_number' => external_tenant}}.to_json) }
+
+  context "with tenancy enforcement" do
+    after { controller.send(:set_current_tenant,  nil) }
+
+    it "get /source with tenant" do
+      headers = { "CONTENT_TYPE" => "application/json", "x-rh-auth-identity" => identity }
+
+      with_modified_env :ENFORCE_TENANCY => "true" do
```

Comment:
Since you are wrapping the entire test in an ENV change, then you can just use the built-in rspec stub_const instead of introducing a gem.  For example,

```
context "with tenancy enforcement" do
  before { stub_const("ENFORCE_TENANCY", true) }
  ...
end
```